enterprise/backend/src/metabase_enterprise/cache/strategies.clj

```diff
@@ -28,64 +26,52 @@
                 ^:internal
-                [:invalidated-at [:maybe some?]]]]]])
+                [:invalidated-at {:optional true} some?]]]]])
 
 ;;; Querying DB
 
-(defn granular-duration-hours
-  "Returns the granular cache duration (in hours) for a card. On EE, this first checking whether there is a stored value
-   for the card, dashboard, or database (in that order of decreasing preference). Returns nil on OSS."
-  [card dashboard-id]
-  (or (:cache_ttl card)
-      (when dashboard-id
-        (t2/select-one-fn :cache_ttl [:model/Dashboard :cache_ttl] :id dashboard-id))
-      (t2/select-one-fn :cache_ttl [:model/Database :cache_ttl] :id (:database_id card))))
-
 (defenterprise-schema granular-cache-strategy :- [:maybe (CacheStrategy)]
   "Returns the granular cache strategy for a card."
   :feature :cache-granular-controls
   [card dashboard-id]
-  (when (public-settings/enable-query-caching)
-    (let [qs   (for [[i model model-id] [[1 "question"   (:id card)]
-                                         [2 "dashboard"  dashboard-id]
-                                         [3 "database"   (:database_id card)]
-                                         [4 "root"       0]]
-                     :when              model-id]
-                 {:from   [:cache_config]
-                  :select [:id
-                           [[:inline i] :ordering]]
-                  :where  [:and
-                           [:= :model model]
-                           [:= :model_id model-id]]})
-          q    {:from     [[{:union-all qs} :unused_alias]]
-                :select   [:id]
-                :order-by :ordering
-                :limit    1}
-          item (t2/select-one :model/CacheConfig :id q)]
-      (if item
-        (-> (:strategy (api.cache/row->config item))
-            (m/assoc-some :invalidated-at (t/max (:invalidated_at item)
-                                                 (:cache_invalidated_at card))))
-        (when-let [ttl (granular-duration-hours card dashboard-id)]
-          {:type     :duration
-           :duration ttl
-           :unit     "hours"})))))
+  (let [qs   (for [[i model model-id] [[1 "question"   (:id card)]
+                                       [2 "dashboard"  dashboard-id]
+                                       [3 "database"   (:database_id card)]
+                                       [4 "root"       0]]
+                   :when              model-id]
+               {:from   [:cache_config]
+                :select [:id
+                         [[:inline i] :ordering]]
+                :where  [:and
+                         [:= :model [:inline model]]
+                         [:= :model_id model-id]]})
+        q    {:from     [[{:union-all qs} :unused_alias]]
+              :select   [:id]
+              :order-by :ordering
+              :limit    [:inline 1]}
+        item (t2/select-one :model/CacheConfig :id q)]
+    (when item
+      (:strategy (api.cache/row->config item card)))))
```

Comment:
This function name is a bit confusing now that it takes an optional card as well. What do you think about keeping the old definition of `row->config` and having something like this?

```clojure
(:strategy (-> (api.cache/row->config item) 
               (api.cache/with-latest-invalidation card))))
```

Reply:
I tried and it makes usage a bit more wordy and easier to slip - since you have to pass `(:invalidated_at row)` anyways, it looks complete after just a row. I think I'll leave it like that.

Reply:
An optional argument is also very easy to forget, especially when the function is named like it only takes a single parameter. While it's more verbose, a second method at least clarifies the purpose of argument and highlights why its required. 

As it stands the function name is very misleading, so at the least I think you should try find a better name if keeping as is. Perhaps `row+card->config`, or even separate 1-arity and 2-arity methods like `base-config` and `card-config`.

Another idea that occurs to me, and is perhaps even cleaner, without extra verbosity, is to replace the `:strategy` keyword with a function that does this merge. This comment is on an outdated diff so I'll go into more detail on the latest  one.